### PR TITLE
operator-plane: localize security under adapter

### PR DESCRIPTION
**Pull request by OrionArchitekton**
## Summary
- reduce `operator-plane` to an adapter-owned boundary by relocating the repo-root `security/` package under `adapter/`
- move quarantine config ownership from `ui/` to `config/`
- add a repo-shape test that enforces the adapter-only boundary and update affected imports/tests

## What Changed
- renames `security/*` to `adapter/security/*`
- rewrites runtime and test imports from `security.*` to `adapter.security.*`
- updates packaging in `pyproject.toml` so the repo no longer exposes a top-level `security*` package
- renames `ui/quarantine_config.json` to `config/quarantine_config.json`
- adds `tests/test_adapter_boundary.py`
- updates `README.md` and `golden-trace.yml` to reflect the split operator topology and boundary checks

## Verification
- `TMPDIR=/tmp pytest tests/test_adapter_boundary.py -v`
- `TMPDIR=/tmp pytest tests/test_adapter_boundary.py tests/test_quarantine_config_api.py tests/test_enforcement.py tests/test_security/test_capability_firewall.py tests/test_security/test_rbac_enforcement.py tests/test_security/test_receipt_first.py -k 'not DenialReceipts' -v`

## Residual Gaps
- JWT-specific tests still require `PyJWT` in the environment; they were not revalidated here
- three `DenialReceipts` tests remain failing because their Chronicle patch target does not match the exercised middleware path; that looks pre-existing and is not addressed in this refactor

### CODE REVIEW

File: adapter/server.py [hunk 1]
```diff
@@ -1051,7 +1052,7 @@ def get_quarantine_config_path() -> Path:
     env_path = os.getenv("OPERATOR_PLANE_QUARANTINE_CONFIG_PATH", "").strip()
     if env_path:
         return Path(env_path)
-    return Path(__file__).parent.parent / "ui" / "quarantine_config.json"
+    return Path(__file__).parent.parent / "config" / "quarantine_config.json"
```
[the remark this answers is not in the document]
Comment (OrionArchitekton) [reply adapter/server.py]: > "Consider centralizing this path resolution"

Addressed in `6ab9f0f`: added `adapter/config_paths.py` and rewired the quarantine config path through the shared helper.
